dn.go

```diff
@@ -208,6 +208,27 @@ func (d *DN) AncestorOf(other *DN) bool {
 	return true
 }
 
+// String returns string representation of the DN Object with required
+// escaping of spaces and formatting applied
+func (d DN) String() string {
+	var buffer bytes.Buffer
+
+	for i := range d.RDNs {
+		if i > 0 {
+			buffer.WriteString(",")
+		}
+		for j := range d.RDNs[i].Attributes {
+			if j > 0 {
+				buffer.WriteString("+")
+			}
+			buffer.WriteString(d.RDNs[i].Attributes[j].Type)
```

Comment:
where is escaping done?

Reply:
Value inspection and escaping (RFC 2253, 4514) is not implemented.  Only canonical representation with spaces removed from [=,+] was done. I can add value inspection and escaping.

Reply:
@liggitt I have added the changes for handing escaping of attribute values as per RFC 4512, 4514. Implementation is generic at DN level and can be used easily.